iceoryx_posh/include/iceoryx_posh/popo/untyped_publisher.hpp

```diff
@@ -25,9 +25,13 @@ namespace iox
 {
 namespace popo
 {
-template <typename base_publisher_t = BasePublisher<>>
+template <typename H = mepoo::NoCustomHeader, typename base_publisher_t = BasePublisher<>>
 class UntypedPublisherImpl : public base_publisher_t
 {
+    static_assert(!std::is_const<H>::value, "The custom header must not be const.");
+    static_assert(!std::is_reference<H>::value, "The custom header must not be a reference.");
+    static_assert(!std::is_pointer<H>::value, "The custom header must not be a pointer.");
```

Comment:
I think it has to be a class? If so, is_class might be better and would make the pointer and reference checks obsolete I think. If we want to have e.g. primitive types as header we cannot use it though.

Reply:
At the moment it would be possible. Don't know it the restriction would harm, but since the code is already there, I would keep it as is.

Reply:
Sure. Just asking what the header can be configured to be. With what you have it can be any non-const value type which makes sense.